### ISSUE TITLE
Flaky test: TestRunAddHost

### ISSUE THREAD

**Issue by AkihiroSuda** [containerd/nerdctl#1127]
```
=== CONT  TestRunAddHost
    run_linux_test.go:108: assertion failed: error is not nil: host was not added, found 0: # <nerdctl>
        127.0.0.1	localhost localhost.localdomain
        :1		localhost localhost.localdomain
        10.0.0.1        test1 test
        10.4.0.140      07fc66977787 alpine-07fc6
        10.4.0.141      ca0911dbb51c alpine-ca091
        127.0.0.1       localhost localhost.localdomain
        :1              localhost localhost.localdomain
        # </nerdctl>
--- FAIL: TestRunAddHost (0.96s)
```

**Comment by manugupt1** [comment]
This seems very specific to the CI.

I can see the failures in https://github.com/containerd/nerdctl/pull/1137

I can't reproduce in either ubuntu (GNOME boxes) / fedora 36 even after 100 runs.

This is the command that I used.

```
 go test -exec sudo -v ./cmd/nerdctl/... -run TestRunAddHost -count=100^C
```

**Comment by AkihiroSuda** [comment]
The test expects `nerdctl run --rm --add-host test:10.0.0.1 --add-host test1:10.0.0.1 alpine cat /etc/hosts`  to generate a file like this
```
# <nerdctl>
127.0.0.1       localhost localhost.localdomain
:1              localhost localhost.localdomain
10.0.0.1        test
10.0.0.1        test1
10.4.0.9        d7c0252b581c alpine-d7c02
# </nerdctl>
```

But occasionally the two lines for `10.0.0.1` are merged into ` 10.0.0.1        test1 test` (seems legit, but should not happen occasionally)

**Comment by AkihiroSuda** [comment]
Looks like the localhost lines are written twice in the failed case, so something seems calling the hosts updater twice
